packages/web3-eth-ens/test/integration/ens.events.test.ts

```diff
@@ -0,0 +1,186 @@
+
+	type ResolverContract = Contract<typeof PublicResolverAbi>;
+
+	let Resolver: ResolverContract;
+	let setEnsResolver: ResolverContract;
+	let getEnsResolver: ResolverContract;
+
+	let sendOptions: PayableTxOptions;
+
+	const domain = 'test';
+	const node = namehash('resolver');
+	const label = sha3('resolver') as string;
+
+	const web3jsName = 'web3js.test';
+
+	const ttl = 3600;
+
+	let accounts: string[];
+	let ens: ENS;
+	let defaultAccount: string;
+	let accountOne: string;
+
+	const ZERO_NODE: Bytes = '0x0000000000000000000000000000000000000000000000000000000000000000';
+	const addressOne: Address = '0x0000000000000000000000000000000000000001';
+
+	beforeAll(async () => {
+		accounts = await getSystemTestAccounts();
+
+		[defaultAccount, accountOne] = accounts;
+
+		sendOptions = { from: defaultAccount, gas: '10000000' };
+
+		const Registry = new Contract(ENSRegistryAbi, undefined, {
+			provider: getSystemTestProvider(),
+		});
+
+		const NameWrapper = new Contract(NameWrapperAbi, undefined, {
+			provider: getSystemTestProvider(),
+		});
+
+		Resolver = new Contract(PublicResolverAbi, undefined, {
+			provider: getSystemTestProvider(),
+		});
+
+		registry = await Registry.deploy({ data: ENSRegistryBytecode }).send(sendOptions);
+
+		nameWrapper = await NameWrapper.deploy({ data: NameWrapperBytecode }).send(sendOptions);
+
+		resolver = await Resolver.deploy({
+			data: PublicResolverBytecode,
+			arguments: [
+				registry.options.address as string,
+				nameWrapper.options.address as string,
+				accountOne,
+				defaultAccount,
+			],
+		}).send(sendOptions);
+
+		await registry.methods.setSubnodeOwner(ZERO_NODE, label, defaultAccount).send(sendOptions);
+		await registry.methods
+			.setResolver(node, resolver.options.address as string)
+			.send(sendOptions);
+		await resolver.methods.setAddr(node, addressOne).send(sendOptions);
+
+		await registry.methods
+			.setSubnodeOwner(ZERO_NODE, sha3(domain) as string, defaultAccount)
+			.send(sendOptions);
+
+		const clientUrl = getSystemTestProvider();
+		let provider;
+		if (isIpc) provider = new ENS.providers.IpcProvider(clientUrl);
+		else if (isWs) provider = new ENS.providers.WebsocketProvider(clientUrl);
+		else provider = new ENS.providers.HttpProvider(clientUrl);
+
+		ens = new ENS(registry.options.address, provider);
+
+		const block = await getBlock(ens, 'latest', false, DEFAULT_RETURN_FORMAT);
+		const gas = block.gasLimit.toString();
+
+		// Increase gas for contract calls
+		sendOptions = {
+			...sendOptions,
+			gas,
+		};
+	});
+
+	afterAll(async () => {
+		if (isSocket) {
+			await closeOpenConnection(ens);
+			// @ts-expect-error @typescript-eslint/ban-ts-comment
+			await closeOpenConnection(ens?._registry?.contract);
+			await closeOpenConnection(getEnsResolver);
+			await closeOpenConnection(setEnsResolver);
+			await closeOpenConnection(registry);
+			await closeOpenConnection(resolver);
+			await closeOpenConnection(nameWrapper);
+		}
```

Comment:
I think there is no need to check if `isSocket` here because the whole test is skipped if it is not. This is according to `describeIf(isSocket)` above. Right?
```suggestion
	await closeOpenConnection(ens);
	// @ts-expect-error @typescript-eslint/ban-ts-comment
	await closeOpenConnection(ens?._registry?.contract);
	await closeOpenConnection(getEnsResolver);
	await closeOpenConnection(setEnsResolver);
	await closeOpenConnection(registry);
	await closeOpenConnection(resolver);
	await closeOpenConnection(nameWrapper);

```

Reply:
right, thanks